src/govuk/components/accordion/accordion.js

```diff
@@ -93,49 +105,78 @@ Accordion.prototype.initSectionHeaders = function () {
 
   // Copy existing span element to an actual button element, for improved accessibility.
   var $button = document.createElement('button')
   $button.setAttribute('type', 'button')
-  $button.setAttribute('id', this.moduleId + '-heading-' + (index + 1))
   $button.setAttribute('aria-controls', this.moduleId + '-content-' + (index + 1))
 
+  // Create show / hide icons with text.
+  var $showIcons = document.createElement('span')
+  $showIcons.classList.add(this.sectionShowHideToggleClass)
+
+  // Add pause (with a comma) after heading for assistive technology.
+  // Example: [heading]Section A ,[pause] Show this section.
+  // https://accessibility.blog.gov.uk/2017/12/18/what-working-on-gov-uk-navigation-taught-us-about-accessibility/
+  var $srPause = document.createElement('span')
+  $srPause.classList.add('govuk-visually-hidden')
+  $srPause.innerHTML = ', '
+
+  // Wrapper of heading to receive focus state design
+  // ID set to allow the heading alone to be referenced without "this section"
+  var $wrapperFocusHeading = document.createElement('span')
+  $wrapperFocusHeading.classList.add(this.sectionHeadingClassFocusWrapper)
+  $wrapperFocusHeading.setAttribute('id', this.moduleId + '-heading-' + (index + 1))
+
+  // Build additional wrapper for toggle text, place icon before wrapped text.
+  var $wrapperToggleText = document.createElement('span')
+  var $icon = document.createElement('span')
+  $icon.classList.add(this.upChevronIconClass)
+  $showIcons.appendChild($icon)
+  $wrapperToggleText.classList.add(this.sectionShowHideTextClass)
+  $showIcons.appendChild($wrapperToggleText)
+
   // Copy all attributes (https://developer.mozilla.org/en-US/docs/Web/API/Element/attributes) from $span to $button
   for (var i = 0; i < $span.attributes.length; i++) {
     var attr = $span.attributes.item(i)
-    $button.setAttribute(attr.nodeName, attr.nodeValue)
-  }
-
-  $button.addEventListener('focusin', function (e) {
-    if (!$headerWrapper.classList.contains($module.sectionHeaderFocusedClass)) {
-      $headerWrapper.className += ' ' + $module.sectionHeaderFocusedClass
+    // Add all attributes but not ID as this is being add directly on $wrapperFocusHeading
+    if (attr.nodeName !== 'id') {
+      $button.setAttribute(attr.nodeName, attr.nodeValue)
     }
-  })
+  }
 
-  $button.addEventListener('blur', function (e) {
-    $headerWrapper.classList.remove($module.sectionHeaderFocusedClass)
-  })
+  $heading.removeChild($span)
+  $heading.appendChild($button)
+  $button.appendChild($wrapperFocusHeading)
+  $button.appendChild($srPause)
+  // span could contain HTML elements (see https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content)
+  $wrapperFocusHeading.innerHTML = $span.innerHTML
 
+  // If summary content exists add to DOM in correct order
   if (typeof ($summary) !== 'undefined' && $summary !== null) {
-    $button.setAttribute('aria-describedby', this.moduleId + '-summary-' + (index + 1))
```

Comment:
If we remove this, we're no longer linking the button to the summary text. That might be okay as the summary is now inside the button but we should probably re-test on screen readers to see if the summary is still announced -  I think you were anticipating this @chris-gds?

If we do retain this change, we should also delete the id from the summary, I don't think it serves any other purpose.

Reply:
>we're no longer linking the button to the summary 

Exactly, as the summary text is now _inside_ the button the additional `aria-describedby` is not longer required as it's not needed to be referenced by something outside.  The heading + summary naturally is announced should be present.

> re-test on screen readers to see if the summary is still announced - I think you were anticipating this @chris-gds?

Yes, there's a fresh round of testing that's needed to just make sure this works as intended.

> If we do retain this change, we should also delete the id from the summary, I don't think it serves any other purpose.

This is a good point 🤔  - as we're updating the template (mentioned below might need a talk) to add `govuk-heading-` this can be removed as well.